### PR TITLE
[MM-16772] Migrate Team.RemoveMember to Sync by default

### PR DESCRIPTION
**Pull request by AGMETEOR**
Summary
This PR migrates Team.RemoveMember in the Store layer to be sync by default

Ticket Link
[Jira Ticket](https://mattermost.atlassian.net/browse/MM-16772)

Fixes [#11556](https://github.com/mattermost/mattermost-server/issues/11556)

### CODE REVIEW

File: store/storetest/user_store.go [hunk 1]
```diff
@@ -3377,8 +3377,8 @@ func testUserStoreGetProfilesNotInTeam(t *testing.T, ss store.Store) {
 	// Ensure update at timestamp changes
 	time.Sleep(time.Millisecond * 10)
 
-	store.Must(ss.Team().RemoveMember(teamId, u1.Id))
-	store.Must(ss.Team().RemoveMember(teamId, u2.Id))
+	// store.Must(ss.Team().RemoveMember(teamId, u1.Id))
```
Comment (jespino) [inline store/storetest/user_store.go]: You are not removing the users as before. We normally replace the `store.Must` with a call getting at least the error and checking the error with `require.Nil(t, err)`.

Comment (AGMETEOR) [reply store/storetest/user_store.go]: Thank you @jespino , I had commented it out in the meantime as I tried to figure out a suitable replacement for that piece of code. I am going to update the code using your suggestion.

**Comment by AGMETEOR** [comment]
@jespino I am getting this error at the end of the tests `Makefile:413: recipe for target 'test-server' failed`. Any ideas on how to fix this?

**Comment by jespino** [comment]
@AGMETEOR the errors are in the entire log, but for see it in an easy way you can see the "Tests" column in the top of the jenkins page, there you can review the tests one by one easily.

**Comment by jespino** [comment]
@AGMETEOR Another thing, we prefer extra commits and merges than using rebases or amends and  `push -f`. Now I have to review the whole PR again (Not a big deal because is a small one, but still a good idea to follow the policy). At the end, it will be the same because we "Squash and merge" the changes, so it will end up being a single commit.

File: store/storetest/user_store.go [hunk 2]
```diff
@@ -3377,8 +3377,12 @@ func testUserStoreGetProfilesNotInTeam(t *testing.T, ss store.Store) {
 	// Ensure update at timestamp changes
 	time.Sleep(time.Millisecond * 10)
 
-	store.Must(ss.Team().RemoveMember(teamId, u1.Id))
-	store.Must(ss.Team().RemoveMember(teamId, u2.Id))
+	e := ss.Team().RemoveMember(teamId, u1.Id)
+	require.Nil(t, e)
+
+	e = ss.Team().RemoveMember(teamId, u1.Id)
```
Comment (jespino) [inline store/storetest/user_store.go]: You have a typo here, `u1` should be `u2`, probably that is the reason why the tests are failing.

Comment (AGMETEOR) [reply store/storetest/user_store.go]: @jespino thank you for the review and guidance. I have made the changes and I think the tests are now passing just fine. I am requesting for an extra review of my PR as this is my first contribution. :)

Comment (jespino) [reply store/storetest/user_store.go]: Sure, no problem 😊. Welcome to the project! :dancer:

**Comment by levb** [comment]
Please pull and resolve the conflicts

**Comment by AGMETEOR** [comment]
Hello @levb I have resolved the conflicts :)

**Pull request merged**